src/Components/Components/src/Forms/FieldIdentifier.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Uniquely identifies a single field that can be edited. This may correspond to a property on a
+    /// model object, or can be any other named value.
+    /// </summary>
+    public readonly struct FieldIdentifier
```

Comment:
note: since this is 2x pointer and readonly, it's a good optimization to use `in` anywhere it's a parameter to reduce copying.

Reply:
OK, I've done this.

However TBH I don't actually know where the perf tradeoff falls in the WebAssembly case. Mono WebAssembly is like 32-bit .NET, though WebAssembly itself has 64-bit int types. Whether this is faster or slower on Mono WebAssembly may depend on whether they have an optimization to copy-on-pass sufficiently small structs as a single 64-bit value.